presto-main/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -274,4 +279,27 @@ public static String getPlanString(PlanNode planNode, Session session, TypeProvi
     {
         return PlanPrinter.textLogicalPlan(planNode, types, StatsAndCosts.empty(), metadata.getFunctionAndTypeManager(), session, 0);
     }
+
+    public static boolean isAllLowCardinalityGroupByKeys(AggregationNode aggregationNode, PlanNode source, Metadata metadata, Session session, StatsCalculator statsCalculator, TypeProvider types, long count)
+    {
+        // We already checked that it's just simple scan/filter/project
+        while (!(source instanceof TableScanNode)) {
```

Comment:
you can use PlanNodeSearcher here to get the TableScanNode

Reply:
That's too complex for my taste lol. I just want to match Scan/Filter/Project nothing else.